Source/WebCore/platform/cocoa/ContentFilterUnblockHandlerCocoa.mm

```diff
@@ -178,14 +179,13 @@
 {
 #if HAVE(WEBCONTENTRESTRICTIONS)
     if (m_evaluatedURL) {
-        if (!m_wcrBrowserEngineClient)
-            m_wcrBrowserEngineClient = adoptNS([PAL::allocWCRBrowserEngineClientInstance() init]);
-        [m_wcrBrowserEngineClient allowURL:*m_evaluatedURL withCompletion:[decisionHandler](BOOL didAllow, NSError *) {
-            callOnMainThread([decisionHandler, didAllow] {
-                RELEASE_LOG(ContentFiltering, "WebFilterEvaluator %s the unblock request.\n", didAllow ? "allowed" : "did not allow");
+        Ref urlFilter = WebCore::ParentalControlsURLFilter::defaultParentalControlsURLFilter();
+        urlFilter->allowURL(*m_evaluatedURL, [decisionHandler = WTFMove(decisionHandler)](bool didAllow) {
+            callOnMainThread([decisionHandler = WTFMove(decisionHandler), didAllow]() {
```

Comment:
Is it always called on a background thread? If not, we could consider calling `ensureOnMainThread()`

Reply:
I think the problem is the completion handler can be invoked synchronously and we want an asynchronous result (not sure if async result is actually needed but given the function name is `requestUnblockAsync` I kept this pattern as the old implementation cc @aestes).